Sentinel/Classes/Core/PreferencesInfo/Views/OptionToggleView.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  OptionToggleView.swift
+//  Sentinel
+//
+//  Created by Zvonimir Medak on 27.11.2024..
+//
+
+import SwiftUI
+
+struct OptionToggleView: View {
+
+    @State private var value: Bool
+    @State private var errorMessage: String?
+    let title: String
+    let description: String?
+    let onValueChanged: (Bool) -> Void
+    let getter: () -> Bool
+    let hasErrorMessage: (Bool) -> String?
+
+    var body: some View {
+        VStack(spacing: 8) {
+            Toggle(isOn: $value) {
+                Text(title)
+                    .font(.body1Bold)
+                    .frame(maxWidth: .infinity, alignment: .leading)
+            }
+
+            DescriptionView(description: description, errorMessage: errorMessage)
+        }
+        .onChange(of: value) { value in
+            guard let message = hasErrorMessage(value) else {
+                errorMessage = nil
+                onValueChanged(value)
+                return
+            }
+            errorMessage = message
```

Comment:
I'm fine with this, but this can also be extracted into a separate method with descriptive method name.